### PR TITLE
Escape some chars in search parameters

### PR DESCRIPTION
**Pull request by cekk**
This is related to [this post](https://community.plone.org/t/parseerror-when-search-term-contains-parenthesis-in-zcatalog-search/13424).

Is a common problem with ZTCTextIndex and this pr will try to fix the problem with searches made with plone.restapi only.

The regexp right now only remove parenthesis from the value. Do we need to manage also other characters?

### CODE REVIEW

**Review by tiberiuichim — COMMENTED** [review]
I don't think it's safe to do this, see https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS

**Comment by tiberiuichim** [comment]
Oh, sorry, I misread your code. I'll remove my comments

**Comment by cekk** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/tests/test_query_parsers.py [hunk 1]
```diff
@@ -42,6 +42,10 @@ def test_casts_complex_query_values_to_string(self):
         qp = ZCTextIndexQueryParser()
         self.assertEqual({"query": "42"}, qp.parse({"query": 42}))
 
+    def test_strip_parenthesis_from_value(self):
+        qp = ZCTextIndexQueryParser()
+        self.assertEqual("foo bar", qp.parse("foo (bar)"))
```
Comment (tisto) [inline src/plone/restapi/tests/test_query_parsers.py:47]: Nitpicking: maybe add the other parenthesis here as well. Just to document that it works for all of them. A dev who reads this test might wonder about it. Just two more lines :)

**Comment by tisto** [comment]
@cekk LGTM. I am just wondering what other characters we would have to strip that would break zcatalog. Would it be possible to write a set of unit tests to just try what works and what not?

**Comment by tisto** [comment]
@jenkins-plone-org please run jobs

**Comment by cekk** [comment]
@tisto i improved tests but i've found some other edge cases that generate exceptions and i don't know if we want to handle or not:

- **not** (already fixed in my pr and tested)
- single chars £ $ * &, etc: ZCTextIndex returns an error if you set SearchableText with one of the special characters

**Comment by cekk** [comment]
@jenkins-plone-org please run jobs

**Review by tisto — APPROVED** [review]
LGTM. A second review would be nice. @cekk sorry it took me so long to check this PR. Busy days... :)

**Comment by cekk** [comment]
@jenkins-plone-org please run jobs

**Comment by cekk** [comment]
@tisto @tiberiuichim any chance to have this pr merged?

**Comment by davisagli** [comment]
I think this is no longer needed since https://github.com/plone/plone.app.querystring/issues/111 was fixed in plone.app.querystring
